### PR TITLE
Implement remaining Game functions

### PR DESCRIPTION
**Pull request by Lectem**
This PR introduces the usage of the squall library (modified version, based on https://github.com/whoahq/squall ) which re-implements Storm trying to match the original code as much as possible.
This will be useful to implement data structures such as lists, hash tables, etc...

Notably, this makes it much easier to reimplement `D2GameStrc` management functions as it was using `Storm`'s `TSExportTableSyncReuse`.

All *GAME* functions are now implemented (but untested)
This is quite a big PR again, due to using those new types. Sorry for that !

### CODE REVIEW

File: source/D2Game/src/GAME/Game.cpp [hunk 1]
```diff
@@ -2620,192 +2419,74 @@ D2GameStrc* __fastcall GAME_GetGameByClientId(int32_t nClientId)
...
+    {
+        GAMEDATALOCKEDHANDLE hLock;
+        if (D2GameStrc* pGame = gpGameDataTbl_6FD45818->tHashTable.Lock(GetGameHandleFromHashValue(nGameGUID), &hLock, TRUE))
+        {
+            if (gpD2EventCallbackTable_6FD45830 && gpD2EventCallbackTable_6FD45830->pfCloseGame)
+            {
+                D2_ASSERT(!IsBadCodePtr((FARPROC)gpD2EventCallbackTable_6FD45830->pfCloseGame));
+                gpD2EventCallbackTable_6FD45830->pfCloseGame(
+                    pGame->nServerToken,
+                    pGame->bExpansion ? 0x44325850 : 0x44324456,
```
Comment (Necrolis) [inline source/D2Game/src/GAME/Game.cpp:2438]: These two constants are ASCII literals of the product codes, `'D2XP'` &  `'D2DV'`respectively

**Comment by heinermann** [comment]
It's worth noting that the Storm information you have was from me originally but is probably the oldest version of that document. Since I'm using a leaner definition for my binary-equivalent Starcraft source it's probably not the best to use so here's an information dump on where you can get your hands on more Storm related stuff.

* Storm names spreadsheet: https://docs.google.com/spreadsheets/d/1Ms7nQgx_AiMWH3-sDWaeMckChap2NiVxsxbPM8-26m4
* Hellfire: https://github.com/pvpgn/diablo-hellfire/blob/master/src/Storm/H/STORM.H
* `stormd.dll` with all its debug info is also shipped with the PT-BR version of Starcraft.

**Comment by Lectem** [comment]
> binary-equivalent Starcraft source it's probably not the best to use

I realized that a bit too late, but I think it mostly matches with D2's version

> * Storm names spreadsheet: https://docs.google.com/spreadsheets/d/1Ms7nQgx_AiMWH3-sDWaeMckChap2NiVxsxbPM8-26m4

Thanks that can definitely help in the future !

> `stormd.dll` with all its debug info is also shipped with the PT-BR version of Starcraft.

Oh we do have one that shipped with D2 too (from the D2 Lod beta)!
I'd probably need to spend some time to cross check everything at some point...

**Review by Mw95 — APPROVED** [review]
Couldn't see any obvious error

**Pull request merged**